apps/builder/app/builder/shared/tree/use-editable.tsx

```diff
@@ -0,0 +1,65 @@
+import {
+  useRef,
+  useEffect,
+  type KeyboardEventHandler,
+  type KeyboardEvent,
+  type FocusEvent,
+} from "react";
+
+export const useEditable = ({
+  isEditing,
+  onChangeValue,
+}: {
+  isEditing: boolean;
+  onChangeValue: (value: string) => void;
+}) => {
+  const elementRef = useRef<HTMLDivElement>(null);
+  const lastValueRef = useRef<string>("");
+  const getValue = () => elementRef.current?.textContent ?? "";
+  const element = elementRef.current;
+
+  useEffect(() => {
+    if (element === null) {
+      return;
+    }
+
+    if (isEditing) {
+      element.setAttribute("contenteditable", "plaintext-only");
+      requestAnimationFrame(() => {
+        element.focus();
+        getSelection()?.selectAllChildren(element);
+        lastValueRef.current = getValue();
+      });
+      return;
+    }
+
+    element.removeAttribute("contenteditable");
+  }, [isEditing, element]);
```

Comment:
there is no need to have ref in a dependency, linter allows and its meant to not be needed, you can just use elementRef directly inside the effect, just don't create a const element variable